### PR TITLE
deb-pkg: add acrn-hypervisor and acrn-board-inspector into debian pac…

### PR DESCRIPTION
**Pull request by hufenglin**
…kage

 1. "make clean && make BOARD=nuc11tnbi5  SCENARIO=shared" will generate the acrn debian package.
 2. "make clean && make board_inspector" will generate the acrn board_inspector debian package.

Tracked-On: #6688
Signed-off-by: Hu Fenglin <fenglin.hu@intel.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -99,6 +100,8 @@ define install_acrn_debug
 endef
 
 HV_MAKEOPTS := -C $(T)/hypervisor BOARD=$(BOARD) SCENARIO=$(SCENARIO) HV_OBJDIR=$(HV_OUT) RELEASE=$(RELEASE)
+board_inspector:
+	@python3 hypervisor/scripts/gen_acrn_deb.py $(ROOT_OUT) $(FULL_VERSION) board_inspect
```
Comment (junjiemao1) [inline Makefile:104]:
```suggestion
	@python3 hypervisor/scripts/gen_acrn_deb.py $(ROOT_OUT) $(FULL_VERSION) board_inspector
```

Comment (junjiemao1) [reply Makefile:104]: Is `gen_acrn_deb.py` designed to be able to generate all deb packages (including DM and tools)? If so, it does not look wise to put it under `hypervisor/scripts`.

File: hypervisor/scripts/gen_acrn_deb.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        DEB_MODE = sys.argv[3]
+    else:
+        BOARD = sys.argv[1]
+        SCENARIO = sys.argv[2]
+        BUILD_DIR = sys.argv[3]
+        HV_VERSION = sys.argv[4]
+        DEB_MODE = 'no'
+
+    cur_dir = os.getcwd()
+    if DEB_MODE != 'board_inspect':
```
Comment (junjiemao1) [inline hypervisor/scripts/gen_acrn_deb.py:143]: It seems Python is not a good choice for this script as most of your code are invoking external tools via the shell. Why not using a shell script instead?

File: Makefile [hunk 3]
```diff
@@ -99,6 +100,8 @@ define install_acrn_debug
 endef
 
 HV_MAKEOPTS := -C $(T)/hypervisor BOARD=$(BOARD) SCENARIO=$(SCENARIO) HV_OBJDIR=$(HV_OUT) RELEASE=$(RELEASE)
+board_inspector:
+	@python3 misc/packaging/gen_acrn_deb.py $(ROOT_OUT) $(FULL_VERSION) board_inspector
```
Comment (dbkinder) [inline Makefile:104]: This seems odd to call the same ``gen_acrn_deb.py`` script with a different number of positional arguments.  This one explicitly says "board_inspector" but the other one doesn't say "hypervisor", for example.

**Comment by dbkinder** [comment]
In the commit message, "make clean && make BOARD=nuc11tnbi5 SCENARIO=share" will fail (scenario name is misspelled)

File: hypervisor/scripts/gen_acrn_deb.py (hunk 2 above)
Comment (dbkinder) [reply hypervisor/scripts/gen_acrn_deb.py]: Will this be rewritten as a shell script as @junjiemao1 recommends?

File: misc/packaging/acrn-hypervisor.postinst [hunk 4]
```diff
@@ -5,117 +5,64 @@
...
-
-}
-
-
-while :
-do
-    read num
-    if [ $num = 0 ] || [ $num -gt ${scenario_num} ]; then
+ACRNBIN="/boot/acrn.${SCENARIO}.${BOARD}.bin"
+type=$(lsblk -l |awk '$NF == "/" {print $1}')
```
[the remark this answers is not in the document]
Comment (hufenglin) [reply misc/packaging/acrn-hypervisor.postinst:17]: Fixed, use one method to get uuid and partuuid.

File: Makefile (hunk 3 above)
Comment (hufenglin) [reply Makefile]: Fixed.

**Comment by hufenglin** [comment]
> In the commit message, "make clean && make BOARD=nuc11tnbi5 SCENARIO=share" will fail (scenario name is misspelled)

thanks, has been fixed.

**Review by junjiemao1 — APPROVED** [review]

**Pull request merged**